src/streamlink/stream/hls.py

```diff
@@ -266,9 +266,13 @@ def reload_playlist(self):
             self.process_sequences(playlist, sequences)
 
     def _playlist_reload_time(self, playlist: M3U8, sequences: List[Sequence]) -> float:
-        if self.playlist_reload_time_override == "segment" and sequences:
+        if self.playlist_reload_time_override == "segment":
+            if not sequences:
+                return playlist.target_duration
```

Comment:
shouldn't we check if `playlist.target_duration` is not **None** and use `self.playlist_reload_time` if it is None
like it is under this

---

also `return sum([s.segment.duration for s in sequences[-max(1, self.live_edge - 1):]])` from `sequences` is a the same as `"live-edge"` basically a duplicate

Reply:
I was trying to avoid the is-float type check which would be required without the `and sequences` condition.

> `playlist.target_duration` is not **None**

This is technically a greater than zero comparison as [the target duration can't be None](https://github.com/streamlink/streamlink/blob/193c5c6eb9147a56df25de160f4d35d6f230733a/src/streamlink/stream/hls_playlist.py#L264) unless the HLS playlist is invalid. Even checking for greater than zero is rather unnecessary as a value of zero would imply that every segment must be at most zero seconds long, which doesn't make sense.

https://datatracker.ietf.org/doc/html/rfc8216#section-4.3.3.1

----

> duplicate

That can't be avoided because of the different priorities of these checks. The second statement basically means "use the `live-edge` method if everything else fails". And the static value of 15, which is the default value (not sure why it was defined like this), is returned when there are no segments available.